libvast/test/data.cpp

```diff
@@ -404,3 +404,43 @@ TEST(pack / unpack) {
   REQUIRE_EQUAL(unpack(*flatbuffer, x2), caf::none);
   CHECK_EQUAL(x, x2);
 }
+
+TEST(get_if) {
+  // clang-format off
+  auto x = record{
+    {"foo", "bar"},
+    {"baz", record{
+      {"qux", int64_t{42}},
+      {"quux", record{
+        {"quuux", 3.14}
+      }},
+    }},
+  };
+  // clang-format on
+  auto foo = get_if<std::string>(&x, "foo");
+  REQUIRE(foo);
+  CHECK_EQUAL(*foo, "bar");
+  auto invalid = get_if<ip>(&x, "foo");
+  CHECK(!invalid);
+  auto baz = get_if<record>(&x, "baz");
+  CHECK(baz);
+  auto qux = get_if<int64_t>(&x, "baz.qux");
+  REQUIRE(qux);
+  CHECK_EQUAL(*qux, 42);
+  auto quux = get_if<record>(&x, "baz.quux");
+  CHECK(quux);
+  auto quuux = get_if<double>(&x, "baz.quux.quuux");
+  REQUIRE(quuux);
+  CHECK_EQUAL(*quuux, 3.14);
+  auto unknown = get_if<ip>(&x, "foo.baz");
+  CHECK(!unknown);
+}
+
+TEST(get_or) {
+  auto x = record{{"foo", "bar"}};
+  auto fallback = std::string{"fallback"};
+  auto foo = get_or(x, "foo", fallback);
+  CHECK_EQUAL(foo, "bar");
+  auto bar = get_or(x, "bar", fallback);
```

Comment:
Can you also add a test for get_or with a fallback string literal? I think there's a missing overload.

Reply:
Ah, right, I missed the convenience overload. We could additionally provide an overload for cheap-to-copy types. Thoughts?

Reply:
We agreed that we want to do this in a follow-up if we need it. Resolving as-is with the new string_view overload.